university/src/university/StudentFactory.java

```diff
@@ -0,0 +1,14 @@
+package university;
+
+public class StudentFactory implements AbstractFactory<Student> {
```

Comment:
Good solution - please note that the `AbstractFactory` here is kind of redundant - it is simply too specific. You are passing 4 student-based params (not generic ones), so the only thing that can potentially implement that abstract factory is a student-related factory. You don't have to change anything, I just wanted to add these two cents here